Firestore/core/src/local/leveldb_remote_document_cache.cc

```diff
@@ -254,29 +249,29 @@ DocumentMap LevelDbRemoteDocumentCache::GetMatching(
 
       const std::string& contents = it->value();
       tasks.Execute([this, &results, document_key, contents] {
-        MaybeDocument maybe_doc = DecodeMaybeDocument(contents, document_key);
-        if (maybe_doc.is_document()) {
-          results.Insert(Document(maybe_doc));
+        MutableDocument document = DecodeMaybeDocument(contents, document_key);
+        if (document.is_found_document()) {
+          results.Insert(MutableDocument(document));
```

Comment:
Nit: you don't need to call the `MutableDocument` constructor here because `document` is already of the right type.